### ISSUE TITLE
colcon build fails [iron]

### ISSUE THREAD

**Issue by craigblaster** [adityakamath/optical_flow_ros#4]
A fresh checkout in a new workspace on ros2 iron fails to build with the following output:
```
Starting >>> optical_flow_ros
--- stderr: optical_flow_ros
/usr/lib/python3/dist-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
---
Finished <<< optical_flow_ros [12.2s]

Summary: 1 package finished [22.2s]
  1 package had stderr output: optical_flow_ros

```

**Comment by adityakamath** [comment]
hi @craigblaster, looks like the build was fine, it just finished with a stderr output (which is a warning in this case). The node should work despite this warning. 

If you don't want this warning in the future, you will need to downgrade to an earlier version of setuptools. 

Let me know if this works!

**Comment by craigblaster** [comment]
The reason I think the build failed is because if I source the local package and try to run optical_flow_node.py, I get an error:
```
$ colcon build

Starting >>> optical_flow_ros
--- stderr: optical_flow_ros
/usr/lib/python3/dist-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
---
Finished <<< optical_flow_ros [12.3s]

Summary: 1 package finished [22.4s]
  1 package had stderr output: optical_flow_ros
$ source install/local_setup.bash
$ ros2 run optical_flow_ros optical_flow_node.py
No executable found
```
Did I miss a step or make a mistake?

**Comment by craigblaster** [comment]
A bit of progress today. I followed the latest tutorial to create a python package and hit upon the exact same issue. It seems that colcon is not yet compatible with the new way of doing things.
Thanks to https://robotics.stackexchange.com/a/24349/17856 I've disabled this warning and the build now goes ok. I preferred this rather than downgrading setuptools as other have reported breaking their system with the later approach.

In summary, the bug is with colcon, not this repo

**Comment by adityakamath** [comment]
Perfect, thanks for the update @craigblaster. I'm going to try this colcon fix out with my ROS devices (I've downgraded setuptools to get rid of the warning). Closing this issue.
